spec/services/anno_auth_validator_spec.rb

```diff
@@ -0,0 +1,43 @@
+require 'rails_helper'
+require 'webmock/rspec'
+
+RSpec.describe AnnoAuthValidator do
+
+  context 'when authorize is called' do
+    describe 'validate canvas and user' do
+      before(:each) do
+        Rails.application.config.JWT_Password = 'abc123'
+        Rails.application.config.JWT_Canvas_Verification_Url = 'http://testserver.com'
+      end
+
+      it 'check valid JWT' do
+        JWT.stub(:decode).and_return([{"group_id" => "10", "user_id" => "100"}])
+        stub_request(:get, /testserver.com/).
+          to_return(status: 200, body: [{"group_id": "10", "has_access": "true"}].to_json)
+        expect(AnnoAuthValidator.new("Bearer someencryptedtoken", "http://canvas.testserver.com/canvas/12345").authorize).to eq(true)
```

Comment:
You can use `described_class` instead of calling `AnnoAuthValidator` explicitly. This means that if you ever renamed the class you wouldn't have to edit each spec.